tests/templates/test_subroutines/test_select_pauli_rot.py

```diff
@@ -0,0 +1,292 @@
+                "Z",
+                [2, 3],
+                "Only one target wire can",
+            ),
+        ],
+    )
+    def test_SelectPauliRot_error(self, angles, rot_axis, target_wire, msg_match):
+        """Test that proper errors are raised for SelectPauliRot"""
+
+        wires = qml.registers({"control_wires": 2})
+
+        with pytest.raises(ValueError, match=msg_match):
+            qml.SelectPauliRot(
+                angles=angles,
+                control_wires=wires["control_wires"],
+                target_wire=target_wire,
+                rot_axis=rot_axis,
+            )
+
+    @pytest.mark.parametrize(
+        ("angles", "rot_axis"),
+        [
+            (np.array([1, 2, 3, 4, 5, 6, 7, 8]), "Z"),
+            (np.array([12, 2, 3, 4, 5.4, 6, 7, 8]), "X"),
+            (np.array([1, 2, -3, 4, 5, 26, 7, 8]), "Y"),
+        ],
+    )
+    def test_correctness(self, angles, rot_axis):
+        """Tests the correctness of the SelectPauliRot template.
+        This is done by comparing the results with the naive Select(Rotation) decomposition
+        """
+
+        dev = qml.device("default.qubit", wires=4)
+
+        gate = {"Z": qml.RZ, "Y": qml.RY, "X": qml.RX}
+
+        # Check that applying the SelectPauliRot and adjoint(Select) to an state,
+        # does not modify the state
+        qs = qml.tape.QuantumScript(
+            [
+                qml.Hadamard(0),
+                qml.Hadamard(1),
+                qml.SelectPauliRot(
+                    angles, control_wires=[0, 1, 2], target_wire=3, rot_axis=rot_axis
+                ),
+                qml.Select([gate[rot_axis](-angle, 3) for angle in angles], control=[0, 1, 2]),
+                qml.Hadamard(0),
+                qml.Hadamard(1),
+            ],
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        output = dev.execute(tape[0])[0]
+
+        assert np.isclose(1.0, output[0])
+
+    def test_decomposition(self):
+        """Test that the correct gates are added in the decomposition"""
+
+        decomposition = qml.SelectPauliRot.compute_decomposition(
+            np.array([1, 2, 3, 4, 5, 6, 7, 8]), control_wires=range(3), target_wire=3, rot_axis="Z"
+        )
+
+        for gate in decomposition:
+            assert gate.name in ["CNOT", "RZ"]
+
+    @pytest.mark.jax
+    def test_interface_jax(self):
+        """Test that SelectPauliRot works with jax"""
+
+        from jax import numpy as jnp
+
+        angles = [1, 2, 3, 4]
+
+        wires = qml.registers({"control": 2, "target": 1})
+        dev = qml.device("default.qubit", wires=3)
+
+        qs = qml.tape.QuantumScript(
+            [
+                qml.SelectPauliRot(
+                    jnp.array(angles),
+                    control_wires=wires["control"],
+                    target_wire=wires["target"],
+                    rot_axis="X",
+                )
+            ],
+            [qml.state()],
+        )
```

Comment:
If we can put this under a function that simply take in `angles`, we need to rewrite this for every interface and make everything cleaner.